mm2src/coins/z_coin/z_rpc.rs

```diff
@@ -43,6 +55,379 @@ struct CompactBlockRow {
+        // do not use retain, we would have to return an error if we did not find connection by the passed address
+        let pos = connections
+            .iter()
+            .position(|con| con.addr == server_addr)
+            .ok_or(ERRL!("Unknown lightwalletd address {}", server_addr))?;
+        // shutdown_tx will be closed immediately on the connection drop
+        connections.remove(pos);
+        Ok(())
+    }
+
+    /// Moves the lightwalletd servers that fail in a multi request to the end.
+    pub async fn rotate_servers(&self, no_of_rotations: usize) {
+        let mut connections = self.connections.lock().await;
+        connections.rotate_left(no_of_rotations);
+    }
+
+    /// Check if one of the spawned connections is connected.
+    pub async fn is_connected(&self) -> bool {
+        for connection in self.connections.lock().await.iter() {
+            if connection.is_connected().await {
+                return true;
+            }
+        }
+        false
+    }
+
+    pub async fn count_connections(&self) -> usize { self.connections.lock().await.len() }
+}
+
+/// Attempts to process the request (parse url, etc), build up the config and create new lightwalletd connection
+#[cfg(not(target_arch = "wasm32"))]
+#[allow(dead_code)]
+fn spawn_connect(
+    url: &String,
+    event_handlers: Vec<RpcTransportEventHandlerShared>,
+) -> Result<LightwalletdConnection<Channel>, String> {
+    let uri: Uri = try_s!(url.parse());
+    uri.host().ok_or(ERRL!("Couldn't retrieve host from addr {}", url))?;
+
+    Ok(lightwalletd_connect(url.clone(), event_handlers))
+}
+
+/// Builds up the lightwalletd connection, spawns endless loop that attempts to reconnect to the server
+/// in case of connection errors
+#[allow(dead_code)]
+fn lightwalletd_connect(
+    addr: String,
+    event_handlers: Vec<RpcTransportEventHandlerShared>,
+) -> LightwalletdConnection<Channel> {
+    let (shutdown_tx, shutdown_rx) = oneshot::channel::<()>();
+    // todo check working with proto format
+    let responses = Arc::new(AsyncMutex::new(JsonRpcPendingRequests::default()));
+    let rpc_client = Arc::new(AsyncMutex::new(None));
+
+    let connect_loop = connect_loop(addr.clone(), responses.clone(), rpc_client.clone(), event_handlers);
+
+    let connect_loop = select_func(connect_loop.boxed(), shutdown_rx.compat());
+    spawn(connect_loop.map(|_| ()));
+    LightwalletdConnection {
+        addr,
+        rpc_client,
+        shutdown_tx: Some(shutdown_tx),
+        responses,
+        protocol_version: AsyncMutex::new(None),
+    }
+}
+
+#[cfg(not(target_arch = "wasm32"))]
+#[allow(dead_code)]
+async fn connect_loop<T>(
+    addr: String,
+    _responses: JsonRpcPendingRequestsShared,
+    connection_rpc_client: Arc<AsyncMutex<Option<mpsc::Sender<CompactTxStreamerClient<T>>>>>,
+    event_handlers: Vec<RpcTransportEventHandlerShared>,
+) -> Result<(), ()> {
+    let delay = Arc::new(AtomicU64::new(0));
```

Comment:
It's not required to use an atomic variable here.
Please consider using `usize` or `f64` or another primitive type for `delay`.

Reply:
This function relates to multi servers task and lightwalletd [doesn't need this](https://github.com/KomodoPlatform/atomicDEX-API/pull/1438#discussion_r962897774), I went the wrong way. So no need in this implementation, but I got it.